packages/app/src/backup/services/backup-create.service.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { Endpoint, Method, GetBackupDeviceStatusDataState } from "@mudita/pure"
+import { Result, ResultObject } from "App/core/builder"
+import { AppError } from "App/core/errors"
+import { isResponseSuccessWithData, isResponseSuccess } from "App/core/helpers"
+import { BackupError } from "App/backup/constants"
+import { MetadataStore, MetadataKey } from "App/metadata"
+import { CreateDeviceBackup } from "App/backup/types"
+
+// DEPRECATED
+import logger from "App/__deprecated__/main/utils/logger"
+import DeviceService from "App/__deprecated__/backend/device-service"
+import DeviceFileSystemAdapter from "App/__deprecated__/backend/adapters/device-file-system/device-file-system-adapter.class"
+
+export class BackupCreateService {
+  constructor(
+    private deviceService: DeviceService,
+    private deviceFileSystem: DeviceFileSystemAdapter,
+    private keyStorage: MetadataStore
+  ) {}
+
+  public async createBackup(
+    options: CreateDeviceBackup
+  ): Promise<ResultObject<string[] | undefined>> {
+    if (this.keyStorage.getValue(MetadataKey.BackupInProgress)) {
+      return Result.failed(
+        new AppError(BackupError.BackupInProgress, "Backup is in progress")
+      )
+    }
+
+    this.keyStorage.setValue(MetadataKey.BackupInProgress, true)
+
+    const runDeviceBackupResponse = await this.runDeviceBackup()
+
+    // AUTO DISABLED - fix me if you like :)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const filePath = runDeviceBackupResponse.data!
+
+    const backupFile = await this.deviceFileSystem.downloadDeviceFilesLocally(
+      [filePath],
+      options
+    )
+
+    if (!isResponseSuccessWithData(backupFile)) {
+      return Result.failed(
+        new AppError(BackupError.BackupDownloadFailed, "Download backup fails")
+      )
+    }
+
+    // TODO: Moved removing backup logic to OS
+    const removeFile = await this.deviceFileSystem.removeDeviceFile(filePath)
+
+    if (isResponseSuccess(removeFile)) {
+      logger.info("Removing device file during backup locally fails")
+    }
+
```

Comment:
Shouldn't it also clear `MetadataKey.BackupInProgress` value in `keyStorage` when operation failed and succeeded?